File: test_auth.py
from dotenv import load_dotenv
import os
from google.oauth2.service_account import Credentials
from googleapiclient.discovery import build

load_dotenv('supabase/functions/.env', override=True)
load_dotenv('.env.local')
pk = os.environ.get('GOOGLE_SA_PRIVATE_KEY', '').strip('"').strip('\'').replace('\\n', '\n')
email = os.environ.get('GOOGLE_SA_EMAIL')

try:
    creds = Credentials.from_service_account_info({
        'client_email': email,
        'private_key': pk,
        'token_uri': 'https://oauth2.googleapis.com/token'
    }, scopes=['https://www.googleapis.com/auth/drive'])
    drive_service = build('drive', 'v3', credentials=creds)
    fid = os.environ.get('VITE_CATALOG_IMAGES_FOLDER_ID')
    print('Checking folder id:', fid)
    res = drive_service.files().get(fileId=fid, fields='id, name', supportsAllDrives=True).execute()
    print('Found:', res)
except Exception as e:
    print(f'Error: {e}')

File: test_auth.js
import { google } from 'googleapis';
import dotenv from 'dotenv';
dotenv.config({ path: '.env.local' });
dotenv.config({ path: 'supabase/functions/.env', override: true });

async function check() {
  const auth = new google.auth.GoogleAuth({
    credentials: {
        client_email: process.env.GOOGLE_SA_EMAIL,
        private_key: process.env.GOOGLE_SA_PRIVATE_KEY.replace(/\\n/g, '\n'),
    },
    scopes: ['https://www.googleapis.com/auth/drive']
  });
  const drive = google.drive({ version: 'v3', auth });
  try {
     const res = await drive.files.get({ fileId: process.env.VITE_CATALOG_IMAGES_FOLDER_ID });
     console.log('Exists:', res.status);
  } catch(e) {
     console.error('Error:', e.message);
  }
}
check();
